// maci-contracts/build/typechain-types/factories/contracts/interfaces/IAnonAadhaar__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IAnonAadhaar__factory = void 0;
const ethers_1 = require("ethers");
const _abi = [
    {
        inputs: [
            {
                internalType: "uint256",
                name: "nullifierSeed",
                type: "uint256",
            },
            {
                internalType: "uint256",
                name: "nullifier",
                type: "uint256",
            },
            {
                internalType: "uint256",
                name: "timestamp",
                type: "uint256",
            },
            {
                internalType: "uint256",
                name: "signal",
                type: "uint256",
            },
            {
                internalType: "uint256[4]",
                name: "revealArray",
                type: "uint256[4]",
            },
            {
                internalType: "uint256[8]",
                name: "groth16Proof",
                type: "uint256[8]",
            },
        ],
        name: "verifyAnonAadhaarProof",
        outputs: [
            {
                internalType: "bool",
                name: "",
                type: "bool",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
];
class IAnonAadhaar__factory {
    static createInterface() {
        return new ethers_1.Interface(_abi);
    }
    static connect(address, runner) {
        return new ethers_1.Contract(address, _abi, runner);
    }
}
exports.IAnonAadhaar__factory = IAnonAadhaar__factory;
IAnonAadhaar__factory.abi = _abi;
